Gordon360/Services/HousingService.cs

```diff
@@ -0,0 +1,774 @@
+        /// <param name="sess_cde"> The current session code </param>
+        /// <param name="applicationID"> The application ID number of the application to be edited </param>
+        /// <param name="newEditorID"> The student username of the student who is declared to be the editor of this application (retrieved from the JSON from the front end) </param>
+        /// <param name="newApartmentApplicants"> Array of JSON objects providing apartment applicants </param>
+        /// <param name="newApartmentChoices"> Array of JSON objects providing apartment hall choices </param>
+        /// <returns>Returns the application ID number if all the queries succeeded</returns>
+        public int EditApplication(string userID, string sess_cde, int applicationID, string newEditorID, ApartmentApplicantViewModel[] newApartmentApplicants, ApartmentChoiceViewModel[] newApartmentChoices)
+        {
+            CCTEntities1 context = new CCTEntities1();
+
+            IEnumerable<string> editorResult = null;
+
+            SqlParameter appIDParam = new SqlParameter("@APPLICATION_ID", applicationID);
+
+            editorResult = RawSqlQuery<string>.query("GET_AA_EDITOR_BY_APPID @APPLICATION_ID", appIDParam);
+            if (editorResult == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The application could not be found." };
+            }
+            else if (!editorResult.Any())
+            {
+                return -1;
+            }
+
+            string storedEditorID = editorResult.FirstOrDefault();
+
+            if (userID != storedEditorID)
+            {
+                // Return -1 if the current user does not match this application's editor stored in the database
+                return -1;
+            }
+            // Only perform the update if the username of the current user matched the 'EditorUsername' stored in the database for the requested application
+
+
+            //--------
+            // Update applicant information
+
+            IEnumerable<GET_AA_APPLICANTS_BY_APPID_Result> existingApplicantResult = null;
+
+            appIDParam = new SqlParameter("@APPLICATION_ID", applicationID);
+
+            // Get the IDs of the applicants that are already stored in the database for this application
+            existingApplicantResult = RawSqlQuery<GET_AA_APPLICANTS_BY_APPID_Result>.query("GET_AA_APPLICANTS_BY_APPID @APPLICATION_ID", appIDParam);
+            if (existingApplicantResult == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The applicants could not be found." };
+            }
+
+            // List of applicants IDs that are in the array recieved from the frontend but not yet in the database
+            List<ApartmentApplicantViewModel> applicantsToAdd = new List<ApartmentApplicantViewModel>(newApartmentApplicants);
```

Comment:
Rather than repeating the type twice in this declaration, this would be an excellent use case for the `var` keyword.